ttnn/cpp/ttnn/decorators.hpp

```diff
@@ -113,6 +113,16 @@ auto map_execute_on_worker_thread_return_to_launch_op_return(const T&& value) ->
         return value;
     } else if constexpr (std::is_same_v<std::decay_t<decltype(value)>, Tensor>) {
         return {value};
+    } else if constexpr (std::is_same_v<std::decay_t<decltype(value)>, OptionalTensors>) {
+        Tensors output_tensors;
+        auto size = value.size();
+        output_tensors.reserve(size);
+
+        auto dummy_tensor = Tensor();
+        for (uint32_t i = 0 ; i < size; i++) {
+            output_tensors.push_back(value.at(i).value_or(dummy_tensor));
```

Comment:
doesn't make sense to use at(i) here
also it is better to use range for loop.

Reply:
>doesn't make sense to use at(i) here.

I didn't understand the part above. could you please explain the reason mentioned above?

```C++
        auto dummy_tensor = Tensor();
        for (auto& val : value) {
            output_tensors.push_back(val.value_or(dummy_tensor));
        }
```
I changed it to use a range-based for loop.